### DISCUSSION TITLE
Highlighting backslash escape sequences in Python string literals

### DISCUSSION THREAD

**Discussion by Flimm** [xonsh/xonsh#4693] [Q&A] [answered]
I would like xonsh to highlight backslash escape sequences in Python string literals, so that string literals look like this:

![image](https://user-images.githubusercontent.com/355454/156714127-a0dee605-0c3b-4c34-819e-ecced6563eff.png)

I am still learning xonsh. I spent ages trying to understand why this command wasn't giving me the expected result:

![`echo hello | grep `\bhello\b`](https://user-images.githubusercontent.com/355454/156714272-77aaa69b-df4f-4e4f-b456-22784d9141f0.png)

In bash, it gives the expected result of `hello`. Eventually, I figured out that xonsh was interpreting `'\bhello\b'` as a Python string literal, and in Python, that string is equivalent to `'\x08hello\x08'`. If I wanted the backslashes to be interpreted literally, then I needed to use a Python raw literal, like this:

```
$ echo hello | grep r'\bhello\b'
hello
$
```

I think I would have worked this out quicker if the backslash escape sequences in the Python string literal would have been highlighted, the way `pymentize` does by default:

![Screenshot from 2022-03-04 07-58-20](https://user-images.githubusercontent.com/355454/156714942-447c66ba-e463-487c-9836-13e7f81da182.png)

How can I configure xonsh to highlight backslash escape sequences in Python string literals?

**Comment by jnoortheen** [comment, marked as answer]
the highlighting is done by [this pygments lexer](https://github.com/xonsh/xonsh/blob/78091a22f3ee9ce201938952868b5e44704e4616/xonsh/pyghooks.py#L1746)

You could update it to get the desired highlighting. Also please open a PR if you make it work or need help (draft PR) implementing it. So it could benefit the community as well.
